### Statistical Tests for Experiment 3

In [2]:
import scipy.stats as stats

import pandas as pd
import numpy as np

In [46]:
# Data Preprocessing

df = pd.read_csv('../dataset/sets/exp3_sets.csv')
df['Optimality_Ratio'] = df['Path_Length'].apply(lambda x: float(x/49))
df['%_Path_Found'] = df['Path_Length'].apply(lambda x: float((49/x)*100))
df['%_Grid_Explored'] = df['Nodes_Expanded'].apply(lambda x: float((x/2500)*100))
df = df.replace([float('inf'), float('-inf')], float('nan')).dropna()


display(df)

,Experiment,Movement_Model,Algorithm,Grid_Size,Density,Execution_Time,Nodes_Expanded,Path_Length,Optimality_Ratio,%_Path_Found,%_Grid_Explored
0,1,4-dir,BFS,Large(50x50),Large(20%),11.020,1978.0,55.0,1.122449,89.090909,79.12
1,1,4-dir,BFS,Large(50x50),Large(20%),12.145,2020.0,55.0,1.122449,89.090909,80.80
2,1,4-dir,BFS,Large(50x50),Large(20%),10.700,2021.0,52.0,1.061224,94.230769,80.84
3,1,4-dir,BFS,Large(50x50),Large(20%),10.380,2008.0,54.0,1.102041,90.740741,80.32
4,1,4-dir,BFS,Large(50x50),Large(20%),11.095,1977.0,56.0,1.142857,87.500000,79.08
...,...,...,...,...,...,...,...,...,...,...,...
229,3,8-dir,D*_Lite,50x50,Large(20%),7.460,534.0,99.0,2.020408,49.494949,21.36
230,3,8-dir,D*_Lite,50x50,Large(20%),10.620,694.0,101.0,2.061224,48.514851,27.76
234,3,8-dir,D*_Lite,50x50,Large(20%),8.875,540.0,99.0,2.020408,49.494949,21.60
236,3,8-dir,D*_Lite,50x50,Large(20%),6.385,500.0,101.0,2.061224,48.514851,20.00


Statistical Tests

In [43]:
# Define groups
g1 = df[df['Movement_Model'] == '4-dir']['Execution_Time']
g2 = df[df['Movement_Model'] == '8-dir']['Execution_Time']

# Test for Normality (Shapiro-Wilk)
if stats.shapiro(g1).pvalue > 0.05 and stats.shapiro(g2).pvalue > 0.05:
    stat, p = stats.ttest_ind(g1, g2)
    print(f"Independent T-Test: stat={stat:.4f}, p={p:.4f}")
else:
    stat, p = stats.mannwhitneyu(g1, g2)
    print(f"Mann-Whitney U Test for Execution Time: stat={stat:.4f}, p={p:.4f}")

Mann-Whitney U Test for Execution Time: stat=3166.5000, p=0.1014


In [44]:
# Define groups
g1 = df[df['Movement_Model'] == '4-dir']['Nodes_Expanded']
g2 = df[df['Movement_Model'] == '8-dir']['Nodes_Expanded']

# Test for Normality (Shapiro-Wilk)
if stats.shapiro(g1).pvalue > 0.05 and stats.shapiro(g2).pvalue > 0.05:
    stat, p = stats.ttest_ind(g1, g2)
    print(f"Independent T-Test: stat={stat:.4f}, p={p:.4f}")
else:
    stat, p = stats.mannwhitneyu(g1, g2)
    print(f"Mann-Whitney U Test for Node Expansion: stat={stat:.4f}, p={p:.4f}")

Mann-Whitney U Test for Node Expansion: stat=3067.0000, p=0.0531


In [ ]:
# Define groups
g1 = df[df['Movement_Model'] == '4-dir']['%_Path_Found']
g2 = df[df['Movement_Model'] == '8-dir']['%_Path_Found']

# Test for Normality (Shapiro-Wilk)
if stats.shapiro(g1).pvalue > 0.05 and stats.shapiro(g2).pvalue > 0.05:
    stat, p = stats.ttest_ind(g1, g2)
    print(f"Independent T-Test: stat={stat:.4f}, p={p:.4f}")
else:
    stat, p = stats.mannwhitneyu(g1, g2)
    print(f"Mann-Whitney U Test for % of Path Found : stat={stat:.4f}, p={p:.4f}")

Mann-Whitney U Test for Path_Length: stat=7434.0000, p=0.0000


In [51]:
# Define groups
g1 = df[df['Movement_Model'] == '4-dir']['%_Grid_Explored']
g2 = df[df['Movement_Model'] == '8-dir']['%_Grid_Explored']

# Test for Normality (Shapiro-Wilk)
if stats.shapiro(g1).pvalue > 0.05 and stats.shapiro(g2).pvalue > 0.05:
    stat, p = stats.ttest_ind(g1, g2)
    print(f"Independent T-Test: stat={stat:.4f}, p={p:.4f}")
else:
    stat, p = stats.mannwhitneyu(g1, g2)
    print(f"Mann-Whitney U Test for % of Grid Explored : stat={stat:.4f}, p={p:.4f}")

Mann-Whitney U Test for % of Grid Explored : stat=3067.0000, p=0.0531


In [52]:
# Define groups
g1 = df[df['Movement_Model'] == '4-dir']['Optimality_Ratio']
g2 = df[df['Movement_Model'] == '8-dir']['Optimality_Ratio']

# Test for Normality (Shapiro-Wilk)
if stats.shapiro(g1).pvalue > 0.05 and stats.shapiro(g2).pvalue > 0.05:
    stat, p = stats.ttest_ind(g1, g2)
    print(f"Independent T-Test: stat={stat:.4f}, p={p:.4f}")
else:
    stat, p = stats.mannwhitneyu(g1, g2)
    print(f"Mann-Whitney U Test for Optimality Ratio : stat={stat:.4f}, p={p:.4f}")

Mann-Whitney U Test for Optimality Ratio : stat=0.0000, p=0.0000


Statistical Tests Summary

Mann-Whitney U Test for Execution Time: stat=3166.5000, p=0.1014

Mann-Whitney U Test for Node Expansion: stat=3067.0000, p=0.0531

Mann-Whitney U Test for Path_Length: stat=7434.0000, p=0.0000

Mann-Whitney U Test for % of Grid Explored : stat=3067.0000, p=0.0531

Mann-Whitney U Test for Optimality Ratio : stat=0.0000, p=0.0000